### PR TITLE
Features/custom prefix handler

### PR DESCRIPTION
**Pull request by SniffBakaSniff**

### CODE REVIEW

File: Migrations/BotDbContextModelSnapshot.cs
Comment (Ascyt) [inline Migrations/BotDbContextModelSnapshot.cs:1]: I would add this to .gitignore, as this is about the current state of the database, and since the database itself is in .gitignore (afaik it is, either way it should be) this should be too

Comment (Ascyt) [reply Migrations/BotDbContextModelSnapshot.cs:1]: The same with the other files in the Migrations folder

File: Features/PrefixCommand.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            if (string.IsNullOrWhiteSpace(newPrefix))
+            {
+                await ctx.RespondAsync("Prefix cannot be empty.");
+                return;
+            }
+
+            await _databaseService.SetPrefixAsync(ctx.Guild.Id, newPrefix);
+
+            await ctx.RespondAsync(
+                MessageHelpers.GenericEmbed("",$"Prefix successfully updated to `{newPrefix}`.", "#00ffff")
```
Comment (Ascyt) [inline Features/PrefixCommand.cs:32]: I think we have a `GenericSuccessEmbed`? If yes, use it instead, otherwise we should create it, but we can do that in another commit

```suggestion
                MessageHelpers.GenericSuccessEmbed(title:"Prefix updated", description:$"Prefix successfully updated to `{newPrefix}`.")
```

File: Features/PrefixCommand.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+namespace ArtcordAdminBot.Features
+{
+    public class PrefixCommand
+    {
+        private readonly IDatabaseService _databaseService;
+
+        public PrefixCommand(IDatabaseService databaseService)
+        {
+            _databaseService = databaseService;
```
Comment (Ascyt) [inline Features/PrefixCommand.cs:15]: Use using statements, see the other comment I made on the other file for info

File: Program.cs [hunk 3]
```diff
@@ -50,15 +62,13 @@ public static async Task Main(string[] args)
 
             DiscordClient client = builder.Build();
 
-            // We can specify a status for our bot. Let's set it to "playing" and set the activity to "with fire".
-            DiscordActivity status = new("with fire", DiscordActivityType.Playing);
+            DiscordActivity status = new("Subaka", DiscordActivityType.ListeningTo);
```
Comment (Ascyt) [inline Program.cs:65]: lol

File: Services/CustomPrefixResolver.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+using DSharpPlus.Entities;
+using DSharpPlus.Commands.Processors.TextCommands.Parsing;
+
+public class CustomPrefixResolver : IPrefixResolver
+{
+    private readonly IDatabaseService _database;
+
+    public CustomPrefixResolver(IDatabaseService database)
+    {
+        _database = database;
```
Comment (Ascyt) [inline Services/CustomPrefixResolver.cs:11]: I would say use using statements instead of injection, see the other comment I made for more info

File: Features/PrefixCommand.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        //[RequirePermissions(DiscordPermissions.Administrator)]
+        public async Task SetPrefixAsync(CommandContext ctx, string newPrefix)
+        {
+            if (string.IsNullOrWhiteSpace(newPrefix))
+            {
+                await ctx.RespondAsync("Prefix cannot be empty.");
+                return;
+            }
+
+            await _databaseService.SetPrefixAsync(ctx.Guild.Id, newPrefix);
```
Comment (Ascyt) [inline Features/PrefixCommand.cs:29]: I see on another file (CustomPrefixResolver.cs) you're case-insensitive about the prefixes - if that is the case, I would actually set the prefix to a lowercase version of what has been submitted:

```suggestion
            newPrefix = newPrefix.ToLower();
            await _databaseService.SetPrefixAsync(ctx.Guild.Id, newPrefix);
```

File: Features/PrefixCommand.cs
Comment (Ascyt) [inline Features/PrefixCommand.cs:1]: Consider adding some length limitation to the prefix, as it could break things if someone submits a prefix with with 4k+ characters

**Review by Ascyt — CHANGES_REQUESTED** [review]
Good stuff, but check the comments of the review, I would suggest some changes

File: Features/PrefixCommand.cs (hunk 2 above)
Comment (SniffBakaSniff) [reply Features/PrefixCommand.cs:15]: if the database service is using the "using" statement then do i need to use using statements here?

File: Features/PrefixCommand.cs
Comment (SniffBakaSniff) [reply Features/PrefixCommand.cs:1]: that makes since

File: Services/CustomPrefixResolver.cs (hunk 4 above)
Comment (SniffBakaSniff) [reply Services/CustomPrefixResolver.cs:11]: i dont understand

File: Features/PrefixCommand.cs (hunk 2 above)
Comment (SniffBakaSniff) [reply Features/PrefixCommand.cs:15]: idk what im doing >_<

Comment (Ascyt) [reply Features/PrefixCommand.cs:15]: Whenever you access the database, you should use them. If you use them in the database service then you're not gonna need it when you access the database service

**Review by Ascyt — APPROVED** [review]
